__tests__/mutate-errors-to-todos-test.ts

```diff
@@ -0,0 +1,42 @@
+import { buildTodoData } from '@ember-template-lint/todo-utils';
+import '../src/estlint-type-extension.d.ts';
+import { mutateTodoErrorsToTodos } from '../src/mutate-errors-to-todos';
+import { readJson } from './__utils__/read-json';
+
+describe('mutate-errors-to-todos', () => {
+  it('changes only the errors that are also present in the todo map to todos', async () => {
```

Comment:
Here I'm testing that the main role of the function: mutate errors to todos, but only those that are present in the todo dir.

So this function is mutating all error results that **are present** in the todo dir to a todo item in ESLint, but if the error is **not present** in the todo dir, then it should remain unchanged, meaning that it should still be reported as an error in ESLint.

This function in fact doesn't really read the todo dir as this is done in the calling function which then passes the todo map to this function.

So what this test is doing:

1. Take a list of error results.
2. Copy this list but remove the last item. This is the actual todo map that would be read from the todo dir from the calling function and passed to this function. We want to simulate the situation where the last error is actually not present in the todo dir.
3. Run the function.
4. Make sure that all errors are mutated to `todos` except the last one because this error was not present in the list of todos.

Makes sense?

Reply:
Can you map this to a workflow for me? What is the workflow this test models?

Reply:
1. Developers runs ESLint and find 3 errors:

```bash
eslint . --format eslint-formatter-todo

✖ 3 problems (3 errors, 0 warnings, 0 todos)
```

2. Developer decides to make all error a TODO.

```bash
UPDATE_TODO=1 eslint . --format eslint-formatter-todo

✖ 3 problems (0 errors, 0 warnings, 3 todos)
```

3. Developer or other developers then continue working and checking in code, **new** ESLint errors are detected. Developer/CI runs:

```bash
eslint . --format eslint-formatter-todo

/Users/user/dev/eslint-test/fullOfProblems.js
   1:10  todo  'addOne' is defined but never used          no-unused-vars

✖ 4 problems (1 error, 0 warnings, 3 todos)
```

This test models after workflows 2 and 3: check if errors are mutated to TODOS but only those that were previously included in the todo list. New errors must be reported as errors.